### ISSUE TITLE
[Bug] Introduction (tutorial?) quests seem to throw errors about Lua

### ISSUE THREAD

**Issue by SoftwareGuy** [Melledy/Grasscutter#1703]
**Did you look for other closed issues that have the same problem?**

Yes, but there weren't any.

**Describe the bug**

When trying to complete the beginning quests, the server logs some errors in the console about it expecting a `userdata` value type but instead it got a number. Refer to additional context below for how to replicate.

```
22:39:41 <ERROR:SceneScriptManager> [LUA] call trigger failed condition_EVENT_ENTER_REGION_24,emu.grasscutter.scripts.data.ScriptArgs@2038d7b2
org.luaj.vm2.LuaError: script:967 bad argument: userdata expected, got number
        at org.luaj.vm2.LuaValue.argerror(Unknown Source)
        at org.luaj.vm2.LuaValue.checkuserdata(Unknown Source)
        at org.luaj.vm2.lib.jse.JavaMethod.call(Unknown Source)
        at org.luaj.vm2.LuaClosure.execute(Unknown Source)
        at org.luaj.vm2.LuaClosure.call(Unknown Source)
        at emu.grasscutter.scripts.SceneScriptManager.safetyCall(SceneScriptManager.java:374)
        at emu.grasscutter.scripts.SceneScriptManager.callScriptFunc(SceneScriptManager.java:367)
        at emu.grasscutter.scripts.SceneScriptManager.realCallEvent(SceneScriptManager.java:322)
        at emu.grasscutter.scripts.SceneScriptManager.lambda$callEvent$10(SceneScriptManager.java:308)
        at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.base/java.lang.Thread.run(Thread.java:833)
22:39:42 <INFO:Scene> Scene 3 loaded 2 group(s)
22:39:44 <INFO:Scene> Scene 3 loaded 2 group(s)
> quest finish 35200
22:41:12 <INFO:CommandHandler> Finished quest 35200.
22:41:12 <ERROR:SceneScriptManager> [LUA] call trigger failed condition_EVENT_ENTER_REGION_24,emu.grasscutter.scripts.data.ScriptArgs@48360b9b
org.luaj.vm2.LuaError: script:967 bad argument: userdata expected, got number
        at org.luaj.vm2.LuaValue.argerror(Unknown Source)
        at org.luaj.vm2.LuaValue.checkuserdata(Unknown Source)
        at org.luaj.vm2.lib.jse.JavaMethod.call(Unknown Source)
        at org.luaj.vm2.LuaClosure.execute(Unknown Source)
        at org.luaj.vm2.LuaClosure.call(Unknown Source)
        at emu.grasscutter.scripts.SceneScriptManager.safetyCall(SceneScriptManager.java:374)
        at emu.grasscutter.scripts.SceneScriptManager.callScriptFunc(SceneScriptManager.java:367)
        at emu.grasscutter.scripts.SceneScriptManager.realCallEvent(SceneScriptManager.java:322)
        at emu.grasscutter.scripts.SceneScriptManager.lambda$callEvent$10(SceneScriptManager.java:308)
        at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.base/java.lang.Thread.run(Thread.java:833)
```

**Which branch did you use?**

Grasscutter v1.2.3, development brach, commit 5db73d2
Note that this is manually compiled, wanted to keep on 2.8.0 until there's solid feedback that the game is functioning correctly with Grasscutter on 3.0.

**Screenshots**
N/A (see console logs above)

**Additional context**

So since my existing account was fubar'd due to a dungeon not working correctly and getting hard-locked in the dungeon where you are on-rails shooting the dragon, I deleted it and started fresh. Since automatic quests aren't implemented, I manually started the quest with the ID of 35200 "Bird's Eye View - Follow the path". I proceeded towards the marker, then the console started spawning errors about it expecting some userdata value but it got a number instead.

I assume this is simply unimplemented behavior, but just wanted to bring it up so I can "start over" 🙂

**Comment by SoftwareGuy** [comment]
Also, maybe related: when logging in after manually completing those quests via server command line, I get this error:

```
> java.lang.NullPointerException: Cannot invoke "emu.grasscutter.scripts.data.SceneGroup.getSuiteByIndex(int)" because "group" is null
        at emu.grasscutter.scripts.SceneScriptManager.refreshGroup(SceneScriptManager.java:107)
        at emu.grasscutter.game.quest.exec.ExecRefreshGroupSuite.execute(ExecRefreshGroupSuite.java:32)
        at emu.grasscutter.game.quest.QuestSystem.triggerExec(QuestSystem.java:95)
        at emu.grasscutter.game.quest.GameQuest.lambda$start$1(GameQuest.java:92)
        at java.base/java.lang.Iterable.forEach(Iterable.java:75)
        at emu.grasscutter.game.quest.GameQuest.start(GameQuest.java:92)
        at emu.grasscutter.game.quest.GameQuest.rewind(GameQuest.java:178)
        at emu.grasscutter.game.quest.GameMainQuest.rewind(GameMainQuest.java:173)
        at emu.grasscutter.game.quest.QuestManager.onLogin(QuestManager.java:90)
        at emu.grasscutter.game.player.Player.onLogin(Player.java:1163)
        at emu.grasscutter.server.packet.recv.HandlerPlayerLoginReq.handle(HandlerPlayerLoginReq.java:48)
        at emu.grasscutter.server.game.GameServerPacketHandler.handle(GameServerPacketHandler.java:88)
        at emu.grasscutter.server.game.GameSession.handleReceive(GameSession.java:247)
        at emu.grasscutter.server.game.GameSessionManager$1.lambda$handleReceive$0(GameSessionManager.java:72)

```

...do we need to implement a fail-safe?

**Comment by Hartie95** [comment]
latest resources should fix this problem, some others problems are still left, like some missing methods, some hidden quests that are not activated, some missing quest triggers, and some bugs in the scriptLib itself.
I'm currently looking into it, and for now got quests working until the point where you need to kill the slime the second time after getting the element.

**Comment by SoftwareGuy** [comment]
Excellent. I'll give this a shot sometime over the weekend and report back!

**Comment by longhuan1999** [comment]
> latest resources should fix this problem, some others problems are still left, like some missing methods, some hidden quests that are not activated, some missing quest triggers, and some bugs in the scriptLib itself. I'm currently looking into it, and for now got quests working until the point where you need to kill the slime the second time after getting the element.

官方原版在进行元素技能教学任务时，提前或者用错误的方式击杀史莱姆会重新开始当前技能教学任务并重新生成史莱姆，所以可能需要延迟生成下一个技能教学任务所需的史莱姆或者检测史莱姆的状态来判断是否重新开始当前技能教学任务
In the official original version, during the element skill teaching task, killing the slime in advance or in the wrong way will restart the current skill teaching task and regenerate the slime, so it may be necessary to delay the generation of the slime required for the next skill teaching task. or check the status of the slime to determine whether to restart the current skill teaching task

**Comment by SoftwareGuy** [comment]
@longhuan1999 

> killing the slime in advance or in the wrong way will restart the current skill teaching task and regenerate the slime

Is that the same as the screen fade out and then reloads the thing? Because if you leave that area around the slime, it will fade out and put you back in front of the slime too.

**Comment by Hartie95** [comment]
> > latest resources should fix this problem, some others problems are still left, like some missing methods, some hidden quests that are not activated, some missing quest triggers, and some bugs in the scriptLib itself. I'm currently looking into it, and for now got quests working until the point where you need to kill the slime the second time after getting the element.
> 
> 官方原版在进行元素技能教学任务时，提前或者用错误的方式击杀史莱姆会重新开始当前技能教学任务并重新生成史莱姆，所以可能需要延迟生成下一个技能教学任务所需的史莱姆或者检测史莱姆的状态来判断是否重新开始当前技能教学任务 In the official original version, during the element skill teaching task, killing the slime in advance or in the wrong way will restart the current skill teaching task and regenerate the slime, so it may be necessary to delay the generation of the slime required for the next skill teaching task. or check the status of the slime to determine whether to restart the current skill teaching task

In theory the check should already be there, it's just that the rewind for restarting some quests on fail are not implemented yet, so the slime never gets respawned.
For now just be carefull to not kill the slime first, or logout an in again to let the slime respawn.
Also check out the quests branch in my fork, for a work in progress version that allows more quests and dungeons to work when scripts in big world are enabled. 
Though it still needs some more changes and testing, before I will create a merge request back into the main Projekt.

**Comment by SoftwareGuy** [comment]
@Hartie95 Is the correct branch on your fork the `quests` one?

**Comment by SoftwareGuy** [comment]
Continuing that thought, I added quest 35200 and got this in the console after a bunch of other stuff scrolling by:
```
18:34:31 <INFO:LuaSerializer> {data_index=0, room_id=0, id=1005, alias=Paimon, script=Actor/Quest/Q352/Paimon, pos=Q352PaimontoNPC, scene_id=3}
java.lang.RuntimeException: Class cannot be created (missing no-arg constructor): sun.reflect.generics.reflectiveObjects.TypeVariableImpl
        at com.esotericsoftware.reflectasm.ConstructorAccess.get(ConstructorAccess.java:68)
        at emu.grasscutter.scripts.serializer.LuaSerializer.cacheType(LuaSerializer.java:191)
        at emu.grasscutter.scripts.serializer.LuaSerializer.serialize(LuaSerializer.java:139)
        at emu.grasscutter.scripts.serializer.LuaSerializer.serializeList(LuaSerializer.java:97)
        at emu.grasscutter.scripts.serializer.LuaSerializer.serialize(LuaSerializer.java:130)
        at emu.grasscutter.scripts.serializer.LuaSerializer.serialize(LuaSerializer.java:161)
        at emu.grasscutter.scripts.serializer.LuaSerializer.serializeMap(LuaSerializer.java:54)
        at emu.grasscutter.scripts.serializer.LuaSerializer.toMap(LuaSerializer.java:35)
        at emu.grasscutter.game.quest.GameMainQuest.addRewindPoints(GameMainQuest.java:212)
        at emu.grasscutter.game.quest.GameMainQuest.<init>(GameMainQuest.java:80)
        at emu.grasscutter.game.quest.QuestManager.addMainQuest(QuestManager.java:178)
        at emu.grasscutter.game.quest.QuestManager.addQuest(QuestManager.java:197)
        at emu.grasscutter.command.commands.QuestCommand.execute(QuestCommand.java:34)
        at emu.grasscutter.command.CommandMap.lambda$invoke$0(CommandMap.java:310)
        at emu.grasscutter.command.CommandMap.invoke(CommandMap.java:314)
        at emu.grasscutter.Grasscutter.startConsole(Grasscutter.java:342)
        at emu.grasscutter.Grasscutter.main(Grasscutter.java:166)
Caused by: java.lang.NoSuchMethodException: sun.reflect.generics.reflectiveObjects.TypeVariableImpl.<init>()
        at java.base/java.lang.Class.getConstructor0(Class.java:3585)
        at java.base/java.lang.Class.getDeclaredConstructor(Class.java:2754)
        at com.esotericsoftware.reflectasm.ConstructorAccess.get(ConstructorAccess.java:65)
        ... 16 more
18:34:31 <INFO:CommandHandler> Quest 35200 added.
```

Quest 35200 is the intro Paimon quest AFAIK.

**Comment by Hartie95** [comment]
> @Hartie95 Is the correct branch on your fork the `quests` one?

yep, so this one:
https://github.com/Hartie95/Grasscutter/tree/quests



> Continuing that thought, I added quest 35200 and got this in the console after a bunch of other stuff scrolling by:
> 
> ```
> 18:34:31 <INFO:LuaSerializer> {data_index=0, room_id=0, id=1005, alias=Paimon, script=Actor/Quest/Q352/Paimon, pos=Q352PaimontoNPC, scene_id=3}
> java.lang.RuntimeException: Class cannot be created (missing no-arg constructor): sun.reflect.generics.reflectiveObjects.TypeVariableImpl
>         at com.esotericsoftware.reflectasm.ConstructorAccess.get(ConstructorAccess.java:68)
>         at emu.grasscutter.scripts.serializer.LuaSerializer.cacheType(LuaSerializer.java:191)
>         at emu.grasscutter.scripts.serializer.LuaSerializer.serialize(LuaSerializer.java:139)
>         at emu.grasscutter.scripts.serializer.LuaSerializer.serializeList(LuaSerializer.java:97)
>         at emu.grasscutter.scripts.serializer.LuaSerializer.serialize(LuaSerializer.java:130)
>         at emu.grasscutter.scripts.serializer.LuaSerializer.serialize(LuaSerializer.java:161)
>         at emu.grasscutter.scripts.serializer.LuaSerializer.serializeMap(LuaSerializer.java:54)
>         at emu.grasscutter.scripts.serializer.LuaSerializer.toMap(LuaSerializer.java:35)
>         at emu.grasscutter.game.quest.GameMainQuest.addRewindPoints(GameMainQuest.java:212)
>         at emu.grasscutter.game.quest.GameMainQuest.<init>(GameMainQuest.java:80)
>         at emu.grasscutter.game.quest.QuestManager.addMainQuest(QuestManager.java:178)
>         at emu.grasscutter.game.quest.QuestManager.addQuest(QuestManager.java:197)
>         at emu.grasscutter.command.commands.QuestCommand.execute(QuestCommand.java:34)
>         at emu.grasscutter.command.CommandMap.lambda$invoke$0(CommandMap.java:310)
>         at emu.grasscutter.command.CommandMap.invoke(CommandMap.java:314)
>         at emu.grasscutter.Grasscutter.startConsole(Grasscutter.java:342)
>         at emu.grasscutter.Grasscutter.main(Grasscutter.java:166)
> Caused by: java.lang.NoSuchMethodException: sun.reflect.generics.reflectiveObjects.TypeVariableImpl.<init>()
>         at java.base/java.lang.Class.getConstructor0(Class.java:3585)
>         at java.base/java.lang.Class.getDeclaredConstructor(Class.java:2754)
>         at com.esotericsoftware.reflectasm.ConstructorAccess.get(ConstructorAccess.java:65)
>         ... 16 more
> 18:34:31 <INFO:CommandHandler> Quest 35200 added.
> ```
> 
> Quest 35200 is the intro Paimon quest AFAIK.

Do you have the `flat.luas.scenes.full_globals.lua.json` from here? 
https://github.com/akatatsu27/MoarGCResources/tree/main/ScriptSceneData

Does the quest work, or does it fail to complete naturally? if it doesn't, does it work if you do them from 35104 to 35200?

**Comment by SoftwareGuy** [comment]
So far so good. Although the console does throw this at me:
```
18:40:55 <ERROR:QuestManager> Unhandled QuestTrigger QUEST_CONTENT_KILL_MONSTER
```

I assume that's because it didn't know what it needs to do (ie. kill entities to ensure the quest doesn't break or stuff happens). 

I killed the slimes and they didn't trigger the next quest, but I manually issued `quest finish` with the required stuff and it continued onwards. The next quest actually then popped and started after a few of the same messages to the big one in the post before.

**Comment by SoftwareGuy** [comment]
> Do you have the `flat.luas.scenes.full_globals.lua.json` from here? https://github.com/akatatsu27/MoarGCResources/tree/main/ScriptSceneData
> 
No, I don't - going to throw that in now and restart the server. I only just used the Grasscutter_resources one. Is there a more "merged" repo I should be using instead?

> Does the quest work, or does it fail to complete naturally? if it doesn't, does it work if you do them from 35104 to 35200?
The quest works, but the game doesn't trigger the cutscene at the end of them. So I kill the slimes as the instructions say and nothing happens, but as soon as I manually finish that quest, the dialogue starts and i talk to paimon.

One thing to notice is that the enemies won't despawn, which can lead you getting killed during the dialogue. If that happens, the game will hang for a bit until it reloads the map.

**Comment by SoftwareGuy** [comment]
~~Where does big LUA file go (the [flat.luas.scenes.full_globals.lua.json](https://github.com/akatatsu27/MoarGCResources/commit/56cb4fc4bce24fc67bfc8617b2d7c5e8256835c8) one) ?~~

Nvm, dumb moment, goes in resources/ScriptSceneData folder.

**Comment by Hartie95** [comment]
> So far so good. Although the console does throw this at me:
> 
> ```
> 18:40:55 <ERROR:QuestManager> Unhandled QuestTrigger QUEST_CONTENT_KILL_MONSTER
> ```
> 
> I assume that's because it didn't know what it needs to do (ie. kill entities to ensure the quest doesn't break or stuff happens).
> 
> I killed the slimes and they didn't trigger the next quest, but I manually issued `quest finish` with the required stuff and it continued onwards. The next quest actually then popped and started after a few of the same messages to the big one in the post before.

`QUEST_CONTENT_KILL_MONSTER` is currently unused, but I still implemented it, but forgot to tell the QuestManager about it, so this warning can be ignored



> > Do you have the `flat.luas.scenes.full_globals.lua.json` from here? https://github.com/akatatsu27/MoarGCResources/tree/main/ScriptSceneData
> 
> No, I don't - going to throw that in now and restart the server. I only just used the Grasscutter_resources one. Is there a more "merged" repo I should be using instead?

not really, this one is the only resource that is used but never got added to the grasscutter resources repository I think.
So for now just put it in resources/ScriptSceneData/

> 
> > Does the quest work, or does it fail to complete naturally? if it doesn't, does it work if you do them from 35104 to 35200?
> > The quest works, but the game doesn't trigger the cutscene at the end of them. So I kill the slimes as the instructions say and nothing happens, but as soon as I manually finish that quest, the dialogue starts and i talk to paimon.
> 
> One thing to notice is that the enemies won't despawn, which can lead you getting killed during the dialogue. If that happens, the game will hang for a bit until it reloads the map.

Interesting, I think in some cases this can even happen on the official server, saw some poeple getting tackled in cutscenes, don't know if it can happen in these starter quests normally though.
Also another thing, I mostly test with `enableScriptInBigWorld` set to true currently, so some parts might only work with that enabled in the config.json

**Comment by SoftwareGuy** [comment]
> Also another thing, I mostly test with enableScriptInBigWorld set to true currently, so some parts might only work with that enabled in the config.json

I have that enabled as well, I like my worlds populated. Otherwise it feels a little... barren. 😃

**Comment by longhuan1999** [comment]
There is a new issue after finish quest 35301 : Unexpected Power - Talk to Paimon. I don't know if it's my problem.
```
<ERROR:GameMainQuest> An error occurred while trying to accept quest.
java.lang.IllegalStateException: stream has already been operated upon or closed
        at java.base/java.util.stream.AbstractPipeline.<init>(AbstractPipeline.java:203)
        at java.base/java.util.stream.ReferencePipeline.<init>(ReferencePipeline.java:96)
        at java.base/java.util.stream.ReferencePipeline$StatelessOp.<init>(ReferencePipeline.java:800)
        at java.base/java.util.stream.ReferencePipeline$3.<init>(ReferencePipeline.java:191)
        at java.base/java.util.stream.ReferencePipeline.map(ReferencePipeline.java:190)
        at emu.grasscutter.scripts.SceneScriptManager.spawnMonstersInGroup(SceneScriptManager.java:336)
        at emu.grasscutter.scripts.SceneScriptManager.refreshGroup(SceneScriptManager.java:162)
        at emu.grasscutter.game.quest.exec.ExecRefreshGroupSuite.execute(ExecRefreshGroupSuite.java:37)
        at emu.grasscutter.game.quest.QuestSystem.triggerExec(QuestSystem.java:92)
        at emu.grasscutter.game.quest.GameQuest.lambda$start$1(GameQuest.java:93)
        at java.base/java.lang.Iterable.forEach(Iterable.java:75)
        at emu.grasscutter.game.quest.GameQuest.start(GameQuest.java:93)
        at emu.grasscutter.game.quest.GameMainQuest.tryAcceptSubQuests(GameMainQuest.java:270)
        at emu.grasscutter.game.quest.QuestManager.triggerEvent(QuestManager.java:263)
        at emu.grasscutter.game.quest.QuestManager.triggerEvent(QuestManager.java:235)
        at emu.grasscutter.game.quest.GameQuest.finish(GameQuest.java:150)
        at emu.grasscutter.game.quest.GameMainQuest.tryFinishSubQuests(GameMainQuest.java:341)
        at emu.grasscutter.game.quest.QuestManager.triggerEvent(QuestManager.java:296)
        at emu.grasscutter.game.quest.QuestManager.triggerEvent(QuestManager.java:235)
        at emu.grasscutter.server.packet.recv.HandlerNpcTalkReq.handle(HandlerNpcTalkReq.java:49)
        at emu.grasscutter.server.game.GameServerPacketHandler.handle(GameServerPacketHandler.java:88)
        at emu.grasscutter.server.game.GameSession.handleReceive(GameSession.java:247)
        at emu.grasscutter.server.game.GameSessionManager$1.lambda$handleReceive$0(GameSessionManager.java:72)
        at io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:174)
        at io.netty.channel.DefaultEventLoop.run(DefaultEventLoop.java:54)
        at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
        at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.base/java.lang.Thread.run(Thread.java:833)
```

**Comment by Hartie95** [comment]
> There is a new issue after finish quest 35301 : Unexpected Power - Talk to Paimon. I don't know if it's my problem.
> 
> ```
> <ERROR:GameMainQuest> An error occurred while trying to accept quest.
> java.lang.IllegalStateException: stream has already been operated upon or closed
>         at java.base/java.util.stream.AbstractPipeline.<init>(AbstractPipeline.java:203)
>         at java.base/java.util.stream.ReferencePipeline.<init>(ReferencePipeline.java:96)
>         at java.base/java.util.stream.ReferencePipeline$StatelessOp.<init>(ReferencePipeline.java:800)
>         at java.base/java.util.stream.ReferencePipeline$3.<init>(ReferencePipeline.java:191)
>         at java.base/java.util.stream.ReferencePipeline.map(ReferencePipeline.java:190)
>         at emu.grasscutter.scripts.SceneScriptManager.spawnMonstersInGroup(SceneScriptManager.java:336)
>         at emu.grasscutter.scripts.SceneScriptManager.refreshGroup(SceneScriptManager.java:162)
>         at emu.grasscutter.game.quest.exec.ExecRefreshGroupSuite.execute(ExecRefreshGroupSuite.java:37)
>         at emu.grasscutter.game.quest.QuestSystem.triggerExec(QuestSystem.java:92)
>         at emu.grasscutter.game.quest.GameQuest.lambda$start$1(GameQuest.java:93)
>         at java.base/java.lang.Iterable.forEach(Iterable.java:75)
>         at emu.grasscutter.game.quest.GameQuest.start(GameQuest.java:93)
>         at emu.grasscutter.game.quest.GameMainQuest.tryAcceptSubQuests(GameMainQuest.java:270)
>         at emu.grasscutter.game.quest.QuestManager.triggerEvent(QuestManager.java:263)
>         at emu.grasscutter.game.quest.QuestManager.triggerEvent(QuestManager.java:235)
>         at emu.grasscutter.game.quest.GameQuest.finish(GameQuest.java:150)
>         at emu.grasscutter.game.quest.GameMainQuest.tryFinishSubQuests(GameMainQuest.java:341)
>         at emu.grasscutter.game.quest.QuestManager.triggerEvent(QuestManager.java:296)
>         at emu.grasscutter.game.quest.QuestManager.triggerEvent(QuestManager.java:235)
>         at emu.grasscutter.server.packet.recv.HandlerNpcTalkReq.handle(HandlerNpcTalkReq.java:49)
>         at emu.grasscutter.server.game.GameServerPacketHandler.handle(GameServerPacketHandler.java:88)
>         at emu.grasscutter.server.game.GameSession.handleReceive(GameSession.java:247)
>         at emu.grasscutter.server.game.GameSessionManager$1.lambda$handleReceive$0(GameSessionManager.java:72)
>         at io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:174)
>         at io.netty.channel.DefaultEventLoop.run(DefaultEventLoop.java:54)
>         at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
>         at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
>         at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
>         at java.base/java.lang.Thread.run(Thread.java:833)
> ```

I will take a look at it later, probably something I overlooked in some newer experimental changes

**Comment by Hartie95** [comment]
> There is a new issue after finish quest 35301 : Unexpected Power - Talk to Paimon. I don't know if it's my problem.
> 
> ```
> <ERROR:GameMainQuest> An error occurred while trying to accept quest.
> java.lang.IllegalStateException: stream has already been operated upon or closed
>         at java.base/java.util.stream.AbstractPipeline.<init>(AbstractPipeline.java:203)
>         at java.base/java.util.stream.ReferencePipeline.<init>(ReferencePipeline.java:96)
>         at java.base/java.util.stream.ReferencePipeline$StatelessOp.<init>(ReferencePipeline.java:800)
>         at java.base/java.util.stream.ReferencePipeline$3.<init>(ReferencePipeline.java:191)
>         at java.base/java.util.stream.ReferencePipeline.map(ReferencePipeline.java:190)
>         at emu.grasscutter.scripts.SceneScriptManager.spawnMonstersInGroup(SceneScriptManager.java:336)
>         at emu.grasscutter.scripts.SceneScriptManager.refreshGroup(SceneScriptManager.java:162)
>         at emu.grasscutter.game.quest.exec.ExecRefreshGroupSuite.execute(ExecRefreshGroupSuite.java:37)
>         at emu.grasscutter.game.quest.QuestSystem.triggerExec(QuestSystem.java:92)
>         at emu.grasscutter.game.quest.GameQuest.lambda$start$1(GameQuest.java:93)
>         at java.base/java.lang.Iterable.forEach(Iterable.java:75)
>         at emu.grasscutter.game.quest.GameQuest.start(GameQuest.java:93)
>         at emu.grasscutter.game.quest.GameMainQuest.tryAcceptSubQuests(GameMainQuest.java:270)
>         at emu.grasscutter.game.quest.QuestManager.triggerEvent(QuestManager.java:263)
>         at emu.grasscutter.game.quest.QuestManager.triggerEvent(QuestManager.java:235)
>         at emu.grasscutter.game.quest.GameQuest.finish(GameQuest.java:150)
>         at emu.grasscutter.game.quest.GameMainQuest.tryFinishSubQuests(GameMainQuest.java:341)
>         at emu.grasscutter.game.quest.QuestManager.triggerEvent(QuestManager.java:296)
>         at emu.grasscutter.game.quest.QuestManager.triggerEvent(QuestManager.java:235)
>         at emu.grasscutter.server.packet.recv.HandlerNpcTalkReq.handle(HandlerNpcTalkReq.java:49)
>         at emu.grasscutter.server.game.GameServerPacketHandler.handle(GameServerPacketHandler.java:88)
>         at emu.grasscutter.server.game.GameSession.handleReceive(GameSession.java:247)
>         at emu.grasscutter.server.game.GameSessionManager$1.lambda$handleReceive$0(GameSessionManager.java:72)
>         at io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:174)
>         at io.netty.channel.DefaultEventLoop.run(DefaultEventLoop.java:54)
>         at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
>         at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
>         at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
>         at java.base/java.lang.Thread.run(Thread.java:833)
> ```

I fixed it in the my new commit. Also it should now autocomplete the unlock the transportation point quests in case you already unlocked it.

You will probably still need these commands though:
* /quest add 35104
* /quest finish 36101 (finishes the hidden quest after dvalin flys over your head)
* /spawn 100135 (spawns the items dvalin leaves behind, give should alternatively also work)

Then for the first flying area with dvalin, teleport out of the domain via hook symbol
* /quest finish 35722
* /quest finish 35725
wait 24h ingame

for the dungeon for amber,lisa and kaeya you need to use the dungeon command for nor
/dungeon 1 (kaeya, will be fully completeable)
/dungeon 1001 (amber, pyro monument won't work, but you can teleport up via `/tp ~ ~+200 ~`, fly to the target and trigger the dialog)
/dungeon 1003 (lisa, will have multiple problems, not working platforms, no working electro monuments and some other possible problems)

Didn't test more, but I think a hidden quest is currently not added that tracks your dungeon progress for those 3 dungeons.

**Comment by longhuan1999** [comment]
> 
> I fixed it in the my new commit. Also it should now autocomplete the unlock the transportation point quests in case you already unlocked it.
> 
> You will probably still need these commands though:
> 
> * /quest add 35104
> * /quest finish 36101 (finishes the hidden quest after dvalin flys over your head)
> * /spawn 100135 (spawns the items dvalin leaves behind, give should alternatively also work)
> 
> Then for the first flying area with dvalin, teleport out of the domain via hook symbol
> 
> * /quest finish 35722
> * /quest finish 35725
>   wait 24h ingame
> 
> for the dungeon for amber,lisa and kaeya you need to use the dungeon command for nor /dungeon 1 (kaeya, will be fully completeable) /dungeon 1001 (amber, pyro monument won't work, but you can teleport up via `/tp ~ ~+200 ~`, fly to the target and trigger the dialog) /dungeon 1003 (lisa, will have multiple problems, not working platforms, no working electro monuments and some other possible problems)
> 
> Didn't test more, but I think a hidden quest is currently not added that tracks your dungeon progress for those 3 dungeons.

好的，我之前就已经注意到了这些并使用“node-red-easy-grasscutters”来自动控制这些缺失的流程实现。非常感谢你和Grasscutters其他相关开发人员的努力。我觉得Dungeons、Quest和Scripts之间相互联系又相对独立，而且 ScriptInBigWorld 似乎已经很久没有维护了，这应该导致修复起来并不容易。
我非常期待能在grasscutter中体验到轻松且完整的剧情，对于我这样不喜欢注意游戏操作和角色培养的玩家来说某些官方剧情任务的难度还是非常大的。或许我不适合玩游戏只适合看剧情吧。
Well, I have noticed these before and used "node-red-easy-grasscutters" to automatically control the missing process implementation. Thank you very much and other relevant developers of Grasscutters for their efforts. I think Dungeons, Quest and Scripts are related and independent, and ScriptInBigWorld seems to have not been maintained for a long time, which should make it difficult to repair.
I am very much looking forward to experiencing a relaxed and complete plot in grasscutter. For a noobs like me, some of the official plot missions are still very difficult. Maybe I'm not suitable for playing games, just for reading stories.

**Comment by KingRainbow44** [comment]
to be fixed by #2173
